modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/segment_edit/SegmentEdit.es.js

```diff
@@ -296,11 +308,22 @@ class SegmentEdit extends Component {
 										type="button"
 									/>
 								</div>
+								<div className="btn-group-item mr-2">
+									<ClayToggle
+										checked={editing}
+										className="toggle-editing"
+										iconOff="view"
+										iconOn="pencil"
+										onChange={this._handleCriteriaEdit}
+									/>
+								</div>
 							</div>
 
 							<div className="btn-group">
 								<div className="btn-group-item">
 									<ClayButton
+										className={(!editing || this._isQueryEmpty()) && "disabled"}
+										disabled={!editing}
```

Comment:
Disabled attribute is meaningless in this case. Perhaps go ahead an add a disabled class inside the ClayButton component when href is present.